### ISSUE TITLE
bug: Screen Reader reads ion-select just as a button

### ISSUE THREAD

**Issue by kensodemann** [ionic-team/ionic-framework#21548]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**
ion-select when used along with ion-select-option is not read by screen reader software like NVDA screen reader properly. It’s just read as a “button”. 

**Expected behavior:**

As per the violation highlighted by the AXE chrome extension, the combobox role (added by ion-select) should have child elements tagged to with role textbox for accessibility tools to work as expected. However, ion-select-option adds html elements with role as “option”

**Related code:**
Sample Usage:

```html
<ion-select id="input-user-address-state" formControlName="state">
  <ion-select-option *ngFor="let state of stateList" [value]="state.value">{{state.name}}</ion-select-option>
</ion-select>
```

Refer to item 1 in the document shared in Slack for further details.
https://ionicteam.slack.com/archives/C7CGXAVTL/p1592319616186200

**Comment by liamdebeasi** [comment]
It appears that the relationship when using `aria-labelledby` is broken in the shadow dom. In this instance, using `aria-labelledby` in the shadow dom and referencing an element outside of the shadow dom does not work. Looks like other people are running into this too: https://developer.paciellogroup.com/blog/2012/07/notes-on-web-components-aria/

I do not know if this is a flaw in the spec, or if this is a bug in browsers. This does not appear to be a bug in the framework. Still investigating.

**Comment by liamdebeasi** [comment]
When fixing this, we should also ensure it resolves the second/related issue described in https://github.com/ionic-team/ionic/issues/21601#issuecomment-649636418.
